plugins/temu.js

```diff
@@ -0,0 +1,125 @@
+﻿var hoverZoomPlugins = hoverZoomPlugins || [];
+hoverZoomPlugins.push({
+    name:'temu',
+    version:'0.1',
+    prepareImgLinks:function (callback) {
+
+        var rawData = null;
+
+        var res = [];
+
+        // sample: https://rewimg-us.kwcdn.com/review-image/1ea26c38d9/4a9d3e35-3033-4d54-9089-ab30fae2e50d_960x1280.jpeg.compressed.jpeg?imageMogr2/auto-orient%7CimageView2/2/w/100/q/70/format/webp
+        //   full: https://rewimg-us.kwcdn.com/review-image/1ea26c38d9/4a9d3e35-3033-4d54-9089-ab30fae2e50d_960x1280.jpeg
+
+        hoverZoom.urlReplace(res,
+            'img[src*="imageMogr2"], img[src*="imageView2"]',
+            /(.*)\?image.*/,
+            '$1'
+        );
+
+        callback($(res), this.name);
+
+        // Hook 'Open' XMLHttpRequests to catch data & metadata associated with pictures or videos displayed
+        // These client/server requests are issued when user scrolls down to retrieve more data
+        // Catched data is stored in sessionStorage for later use by plug-in
+
+        if ($('script.HZTemuOpen').length == 0) { // Inject hook script in document if not already there
+            var hookScript = document.createElement('script');
+            hookScript.type = 'text/javascript';
+            hookScript.text = `if (typeof oldXHROpen !== 'function') { // Hook only once!
+                oldXHROpen = window.XMLHttpRequest.prototype.open;
+                window.XMLHttpRequest.prototype.open = function(method, url, async, user, password) {
+                    // catch responses
+                    this.addEventListener('load', function() {
+                        try {
+                            let data = this.responseText;
+                            // store relevant data as plain text in sessionStorage for later usage by plug-in
+                            if (data.indexOf('cover_image_url') != -1) {
+                                 // update sessionStorage
+                                let storedHookedDataJson = sessionStorage.getItem('HZOpenData') || '[]';
+                                let storedHookedData = JSON.parse(storedHookedDataJson);
+                                // make some room by removing oldest data
+                                while (storedHookedDataJson.length > 1000000) {
```

Comment:
Why 1000000 in particular? Is there any benefit of strong more than a 100 or even a 1000 responses?

Reply:
Each response is approx. 10k, so we can store around 100 responses before starting deleting oldest responses.